### DISCUSSION TITLE
.

### DISCUSSION THREAD

**Discussion by dianfishekqi** [laravel/framework#39072] [Ideas]

**Comment by dennisprudlo** [comment]
In which case would you use a translation key which does not exist?

**Reply by dennisprudlo** [reply]
Have you a specific example? I mean the translation strings are static/fixed in your codebase so usually there is a defined set of translation strings in your application that can be used. Why should somewhere `trans('some.translation.key')` be used when `some.translation.key` does not exist.

**Reply by dennisprudlo** [reply]
But yet again, say the name of the event is `dispatched`. At the moment laravel would show `shipping.events.dispatched` as a fallback if there was no key with the name. You want to show only `dispatched`? In any case the best practice would obviously be to create a corresponding translation string when creating a new event.

But if you really want to check you could use the translators `has` function:
```php
@if (app('translator')->has('shipping.events.'.$eventName))
    {{ trans('shipping.events.'.$eventName) }}
@else
    {{ $eventName }}
@endif
```

**Comment by shaedrich** [comment]
Could you please make the title and text of your discussion more descriptive than just "."
